xapian-core/include/xapian/weight.h

```diff
@@ -29,6 +29,10 @@
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
+#include "registry.h"
+
+#include "../common/stringutils.h"
```

Comment:
If the code which needs `C_isspace()` is in `weight.cc`, then add the header inclusion to `weight.cc` not here.

Including a header in a file where it's not used at all is unhelpful - as your patch currently stands, if a developer modifies `common/stringutils.h` then `make` will have to recompile every source file which directly or indirectly includes `include/xapian/weight.h`, but many of those files probably don't actually use anything from `common/stringutils.h`.